instrumentation/yesod/src/OpenTelemetry/Instrumentation/Yesod.hs

```diff
@@ -249,18 +247,29 @@ openTelemetryYesodMiddleware rr m = do
           }
   case mspan of
     Nothing -> do
-      eResult <- inSpan' (maybe "notFound" (\r -> nameRender rr r) mr) args $ \_s -> do
+      eResult <- inSpan' (maybe "notFound" (nameRender rr) mr) args $ \_s -> do
         catch (Right <$> m) $ \e -> do
-          -- We want to mark the span as an error if it's an InternalError,
-          -- the other HCError values are 4xx status codes which don't
-          -- really count as a server error in OpenTelemetry spec parlance.
-          case e of
-            HCError (InternalError _) -> throwIO e
-            _ -> pure ()
+          when (isInternalError e) $ throwIO e
           pure (Left (e :: HandlerContents))
       case eResult of
         Left hc -> throwIO hc
         Right normal -> pure normal
     Just waiSpan -> do
       addAttributes waiSpan sharedAttributes
-      m
+      withException m $ \ex -> do
```

Comment:
This makes sense, but please add the explanation from the PR here as a comment so we know why we're doing this in future.

Reply:
> so we know why we're doing this in future.

The PR description was taking from the [commit messages](https://github.com/iand675/hs-opentelemetry/pull/121/commits/9610e0fea06c548f3bc159718f30ce73a6f83df3), which means the information will be readily available and immutably attached to the code, which is why I typically prefer that over comments.

I will add the comment too, if you prefer, I just wanted to double check you knew it was in the commit message already too.

Reply:
Commit messages become harder to find over time as the code changes in other ways. I would prefer a comment. (But I am an extremist: https://www.michaelpj.com/blog/2022/04/24/on-commenting-code.html)

Reply:
Done.